lnwallet/channel.go

```diff
@@ -1959,14 +1965,18 @@ func (lc *LightningChannel) restorePendingLocalUpdates(
 					lc.localUpdateLog.htlcCounter))
 			}
 
-			lc.localUpdateLog.appendHtlc(payDesc)
-
+			if lc.localUpdateLog.restoreHtlc(payDesc) {
+				lc.localUpdateLog.htlcCounter++
+				lc.localUpdateLog.logIndex++
+			}
 		case FeeUpdate:
-			lc.localUpdateLog.appendUpdate(payDesc)
-
+			if lc.localUpdateLog.restoreUpdate(payDesc) {
+				lc.localUpdateLog.logIndex++
+			}
 		default:
-			lc.localUpdateLog.appendUpdate(payDesc)
-
+			if lc.localUpdateLog.restoreUpdate(payDesc) {
```

Comment:
Hmm, so then that panic was due to a duplicate update being added, which left the original dangling outside fo the update index?

Reply:
I think the panic was just a coding error I made in my test case